File: src/test/java/common/report/TestCategory.java
package common.report;

import org.junit.experimental.categories.Category;

public enum TestCategory {
    AUTOMATED, SEMI_AUTOMATED, HELPER;

    public static TestCategory getForCategory(final Category category) {
        switch (category.value()[0].getSimpleName()) {
            case "Helper": return TestCategory.HELPER;
            case "SemiAutomated": return TestCategory.SEMI_AUTOMATED;
            default: return TestCategory.AUTOMATED;
        }
    }
}

File: src/test/java/common/util/report/TestCategory.kt
package common.util.report

import org.junit.experimental.categories.Category

enum class TestCategory {
    AUTOMATED, SEMI_AUTOMATED, HELPER;

    companion object {
        @JvmStatic
        fun getForCategory(category: Category): TestCategory {
            return when (category.value[0].simpleName) {
                "Helper" -> HELPER
                "SemiAutomated" -> SEMI_AUTOMATED
                else -> AUTOMATED
            }
        }
    }
}